reference/factorization/par_ilu_kernels.cpp

```diff
@@ -77,6 +79,10 @@ void initialize_row_ptrs_l_u(
         l_row_ptrs[row + 1] = l_nnz + !has_diagonal;
         u_row_ptrs[row + 1] = u_nnz + !has_diagonal;
     }
+
+    // Now, compute the prefix-sum, to get proper row_ptrs for L and U
+    prefix_sum(exec, l_row_ptrs, num_rows + 1);
+    prefix_sum(exec, u_row_ptrs, num_rows + 1);
```

Comment:
Why are you separating the prefix sum calculation from the NNZ counting? In my opinion, that makes the code a bit harder to read and slightly more complicated (it also makes the names `l_nnz` and `u_nnz` a bit misleading since they now only count the NNZ per row).

Reply:
This mainly unifies the reference and OMP implementation, since they both do the same. Additionally, the reference implementation at the moment has a bug when there is an element missing from the diagonal. I should probably add a test that would have failed before.

Reply:
I don't think they should be unified, it would actually be beneficial if they are different (might result in different test behavior, which can show us bugs in either implementation).
If there is a chance of a bug, I agree, adding a test that would cover that case would help us a lot.

Reply:
I reversed the changes, fixed the bug directly and added a corresponding test instead.